subprojects/core/src/main/java/org/gradle/api/internal/file/archive/TarFileTree.java

```diff
@@ -188,7 +208,7 @@ public InputStream open() {
             if (read && file != null) {
                 return GFileUtils.openInputStream(file);
             }
-            if (read || tar.getCurrent() != entry) {
+            if (read) {
```

Comment:
❓ why remove that check?

```suggestion
            if (read || tar.getCurrentEntry() != entry) {
```